src/main/java/com/io/image/manager/controller/ImageController.java

```diff
@@ -111,7 +111,7 @@ public ImageController(ImageService imageService, CacheRecordRepository cacheRep
         // ==============
 
         String normalizedFilename = filename.substring(0, filename.indexOf(".")) + ".jpg";
-
+        String extension =  filename.substring(filename.indexOf(".")+1);
```

Comment:
Probably should be `lastIndexOf` or whatever java has for getting index of last occurence.